source/server-discovery-and-monitoring/tests/integration/minPoolSize-error.yml

```diff
@@ -0,0 +1,79 @@
+# Test SDAM error handling.
+runOn:
+    # failCommand appName requirements
+    - minServerVersion: "4.4"
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "sdam-minPoolSize-error"
+
+data: []
+
+tests:
+  - description: Command error on minPoolSize background creation
+    # Configure the initial monitor handshake to succeed but the
+    # first or second background minPoolSize establishments to fail.
+    failPoint:
+      configureFailPoint: failCommand
+      mode: { skip: 2 }
+      data:
+        failCommands: ["isMaster"]
+        appName: SDAMminPoolSizeError
+        closeConnection: false
+        errorCode: 91 # ShutdownInProgress
+    clientOptions:
+      heartbeatFrequencyMS: 10000
+      appname: SDAMminPoolSizeError
+      minPoolSize: 10
+      serverSelectionTimeoutMS: 100
+    operations:
+      # Wait for monitor to succeed handshake and mark the pool as ready.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolReadyEvent
+          count: 1
+      # Background connection establishment ensuring minPoolSize should fail,
+      # causing the pool to be cleared.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolClearedEvent
+          count: 1
+      # The server should be marked as Unknown as part of this.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: ServerMarkedUnknownEvent
+          count: 1
+      # Executing a command should fail server selection due to not being able
+      # to find the primary.
+      - name: runCommand
+        object: database
+        command_name: ping
+        arguments:
+          command:
+            ping: {}
+        error: true
+      # Disable the failpoint, allowing the monitor to discover the primary again.
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: off
+            data:
+              errorCode: 91 # ShutdownInProgress
+      # Wait for monitor to discover the primary and mark the pool as ready again.
+      - name: waitForEvent
+        object: testRunner
+        arguments:
+          event: PoolReadyEvent
+          count: 1
```

Comment:
This should be `count:2` because we want to wait for the 2nd PoolReadyEvent.

Reply:
fixed